File: Day_7/141. Linked List Cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(head==NULL)
            return false;
        ListNode *x=head, *y=head;

        while(x != NULL  && x->next != NULL){
            x = x->next->next;
            y = y->next;

            if(x==y)
                return true;
        }
        return false;
    }
};

File: Day_7/141. Linked List Cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    if(head === null)
        return false;
    
    let x=head, y=head;
    while(x != null && x.next != null){
        x = x.next.next;
        y = y.next;
        
        if(x===y)
            return true;
    }
    return false;
};
